cns/restserver/internalapi.go

```diff
@@ -143,6 +146,62 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	return
 }
 
+// SyncHostNCVersion will check NC version from NMAgent and save it as host NC version in container status.
+// If NMAgent NC version got updated, CNS will refresh the pending programming IP status.
+func (service *HTTPRestService) SyncHostNCVersion(ctx context.Context, channelMode string, syncHostNCTimeoutMilliSec time.Duration) {
+	var hostVersionNeedUpdateNcList []string
+	service.RLock()
+	for _, containerstatus := range service.state.ContainerStatus {
```

Comment:
If you want, you can extract this earlier portion in a separate function - which returns hostVersionNeedUpdateNcList  which this function uses to call to nma. It's ok if you want to keep it as is

Reply:
Let me keep as it is for now. I can have another PR to make structure better.